### DISCUSSION TITLE
Cover/shutter state & open/close button

### DISCUSSION THREAD

**Discussion by stegaroby** [gicamm/homeassistant-comelit#56] [Q&A]
Hi,
I have 5 shutters, and all of them have the status recognized correctly: real shutter down-->state close, real sutter up --> state open. I have a problem with the open/close button of the cover card: it's inverted! If i have the cover open I can only click on open, because the button cose is disabeled
<img width="200" alt="Screenshot 2023-12-31 134525" src="https://github.com/gicamm/homeassistant-comelit/assets/123300298/9206f55c-cbab-4b86-a989-417d84ba4de4"><img width="200" alt="Screenshot 2023-12-31 134603" src="https://github.com/gicamm/homeassistant-comelit/assets/123300298/c8dc8c52-69c4-45cf-adbb-e4277319cc74"><img width="200" alt="Screenshot 2023-12-31 134814" src="https://github.com/gicamm/homeassistant-comelit/assets/123300298/99603d14-dc8d-416b-92d1-78b522019a76"><img width="200" alt="Screenshot 2023-12-31 134632" src="https://github.com/gicamm/homeassistant-comelit/assets/123300298/e6c56d39-73a9-414c-9a66-1ea68603d577">
I know that I could reverse the wiring of both the buttons and the motor phases, and it would resolve the movement, but the state of the shutters would then be reversed.
Can i solve it through changing in the code?

PS At this moment I have not yet fully integrated home assistant: I am using the Alexa API to activate home automation, and activating the "open" or "close" functions works correctly

Thanks

**Comment by gicamm** [comment]
Hello,
In Home Assistant, by default 0 means closed and 100 means open.
It's weird that the state is correct but the buttons are reversed.

**Reply by stegaroby** [reply]
Hi,
I have installed the extension for 1 year now (end of January 2023), yesterday I tried to reload everything via HACS and now all the shutters are correct.
I solved the problem. 
Thanks
Roby

**Reply by gicamm** [reply]
great!
